### PR TITLE
refactor: make templateImage a property on nativeImage

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Update `nativeImage.templateImage` to be a property on the `nativeImage` prototype.
 
#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Updates `nativeImage.templateImage` to be a property on the `nativeImage` prototype.

### CODE REVIEW

File: docs/api/native-image.md [hunk 1]
```diff
@@ -324,3 +328,9 @@ to explicitly add different scale factor representations to an image. This
 can be called on empty images.
 
 [buffer]: https://nodejs.org/api/buffer.html#buffer_class_buffer
+
+## Properties
+
+### `nativeImage.templateImage`
```
Comment (MarshallOfSound) [inline docs/api/native-image.md]:
```suggestion
### `nativeImage.isTemplateImage`
```

File: docs/api/native-image.md [hunk 2]
```diff
@@ -324,3 +328,9 @@ to explicitly add different scale factor representations to an image. This
...
+
+A `Boolean` property that determines whether the image is a template image.
```
Comment (MarshallOfSound) [inline docs/api/native-image.md]:
```suggestion
A `Boolean` property that determines whether the image is considered a [template image](https://developer.apple.com/documentation/appkit/nsimage/1520017-template).

Please note that this property only has an effect on macOS.
```

File: spec/api-native-image-spec.js [hunk 3]
```diff
@@ -105,6 +105,14 @@ describe('nativeImage module', () => {
     return matchingImage
   }
 
+  describe('templateImage property', () => {
+    it('returns whether the image is a template image', () => {
```
Comment (MarshallOfSound) [inline spec/api-native-image-spec.js]: Can we add tests for the behavior here.  i.e. if you do `nativeImage.createFromPath('MyImage_Template.png`)` then it should be `true`.

If you do `nativeImage.createFromPath('MyImage.png')` it should be `false`.

Only on macOs though

Comment (MarshallOfSound) [reply spec/api-native-image-spec.js]: This tests the setter, but not the getter returns the correct value on the first try.  

```js
nativeImage.createFromPath('MyImage_Template.png') === true
```

Is a test we want here as well

File: atom/common/api/atom_api_native_image.cc [hunk 4]
```diff
@@ -638,8 +638,10 @@ void NativeImage::BuildPrototype(v8::Isolate* isolate,
       .SetMethod("toDataURL", &NativeImage::ToDataURL)
       .SetMethod("isEmpty", &NativeImage::IsEmpty)
       .SetMethod("getSize", &NativeImage::GetSize)
-      .SetMethod("setTemplateImage", &NativeImage::SetTemplateImage)
-      .SetMethod("isTemplateImage", &NativeImage::IsTemplateImage)
+      .SetMethod("_setTemplateImage", &NativeImage::SetTemplateImage)
+      .SetMethod("_isTemplateImage", &NativeImage::IsTemplateImage)
+      .SetProperty("templateImage", &NativeImage::IsTemplateImage,
```
Comment (MarshallOfSound) [inline atom/common/api/atom_api_native_image.cc]: I'm still not a huge fan of this property name, I get we can't call it `isTemplateImage` for deprecation reasons.  But I still feel `nativeImage.templateImage` as a boolean doesn't quite fit.

Maybe `nativeImage.isMacTemplateImage`?

Comment (codebytere) [reply atom/common/api/atom_api_native_image.cc]: hmm yes! this is an unexpectedly annoying naming clash but i definitely agree that `templateImage` might lead the user to make incorrect assumptions about what exactly the property should be set to.  

let's go with `nativeImage.isMacTemplateImage` for now 🤔

File: docs/api/native-image.md [hunk 5]
```diff
@@ -324,3 +327,11 @@ to explicitly add different scale factor representations to an image. This
 can be called on empty images.
 
 [buffer]: https://nodejs.org/api/buffer.html#buffer_class_buffer
+
+## Properties
+
+### `nativeImage.isMacTemplateImage`
```
Comment (MarshallOfSound) [inline docs/api/native-image.md]:
```suggestion
### `nativeImage.isMacTemplateImage` _macOS_
```

☝️ is how we do platform support

**Pull request merged**